core/controller/src/main/scala/whisk/core/controller/Controller.scala

```diff
@@ -202,7 +202,7 @@ object Controller {
 
     // if deploying multiple instances (scale out), must pass the instance number as the
     require(args.length >= 1, "controller instance required")
-    val instance = args(0).toInt
+    val instance = ControllerInstanceId(args(0).trim)
```

Comment:
i think trim is not necessary since you will replace space it in the constructor.

Reply:
I would typically leave this in place, since I think it is good to always trim input strings. But I can remove it since it is technically true that it won't make a difference here.